addons/sale/tests/test_sale_to_invoice.py

```diff
@@ -68,19 +68,40 @@ def setUpClass(cls):
             'default_journal_id': cls.journal_sale.id,
         }
 
+    def check_has_invoices(self, has_invoices):
+        """Check query "Invoices is not set" """
+        domain = [('id', '=', self.sale_order.id), ('invoice_ids', '=', False)]
+        result = self.env['sale.order'].search(domain)
+        self.assertEqual(not result, has_invoices, "Domain [('invoice_ids', '=', False)] doesn't return expected result")
+
+    def test_searching_has_invices(self):
+        """Test "Invoices is not set" query when some lines are not invoiced"""
+        self.sol_prod_order.product_uom_qty = 0
+        # Confirm the SO
+        self.sale_order.action_confirm()
+        self.check_has_invoices(False)
+        # Let's do an invoice with invoiceable lines
+        payment = self.env['sale.advance.payment.inv'].with_context(self.context).create({
+            'advance_payment_method': 'all'
+        })
+        payment.create_invoices()
```

Comment:
Why do you need to go through the wizard ?
`self.sale_order.action_invoice_create()` should be enough :thinking: